### PR TITLE
Hydroponics borg :)

### PR DESCRIPTION
**Pull request by Moltovz**
## About The Pull Request
Added a new Cyborg to aid Hydroponics! The borg module itself is locked behind advanced research, meaning it's not available at round start and rolls out with advanced botany tools. It has a _flash, crowbar, "transport bag" (used to transport seeds, plants, and grafts), gene sheers, a plant analyzer, floragun, cultivator, beaker, secateurs, a hatchet, a spade, and of course- a fire extinguisher_. These borgs **assist** botanist. They can NOT do the job all by themself!
image
image

https://youtu.be/eo_tAPt5vr0

## Why It's Good For The Game
Having extra help in Hydroponics is always appriciated.

## Changelog
Greenlit by oranges.
:cl:
add: A new borg focused around botany (Moltov)
/:cl:

### CODE REVIEW

**Comment by InsaneRed** [comment]
he deserves a flower hat.

**Comment by tralezab** [comment]
whatever "fuckhumans" are, i'm pretty sure i don't want them

hydroponics borg? eeeh... i dunno. even if botany borg is justified, why isn't this all on our most underused model, service borgs?

File: code/game/objects/items/robot/robot_items.dm [hunk 1]
```diff
@@ -1005,6 +1005,58 @@
 		to_chat(user, span_notice("[src] is empty."))
 	return
 
+/////////////////////
+//plant storage bag// //shamelessly copy/pasted organ bag code
+/////////////////////
+
+/obj/item/borg/apparatus/plant_storage
+	name = "hydroponics transport bag"
+	desc = "A container for holding body parts."
```
Comment (Fikou) [inline code/game/objects/items/robot/robot_items.dm:1014]: wrong desc

File: code/game/objects/items/robot/robot_items.dm [hunk 2]
```diff
@@ -1005,6 +1005,58 @@
...
+		stored_flora.plane = FLOAT_PLANE
+		stored_flora.pixel_x = 0
+		stored_flora.pixel_y = 0
+		. += stored_flora
+		bag = mutable_appearance(icon, icon_state = "evidence") // full bag
+	else
+		bag = mutable_appearance(icon, icon_state = "evidenceobj") // empty bag
+	. += bag
+
+/obj/item/borg/apparatus/plant_storage/AltClick(mob/living/silicon/robot/user)
```
Comment (Fikou) [inline code/game/objects/items/robot/robot_items.dm:1049]: ? this is already handled by apparatuses, just cut overlays after it does its job

File: code/modules/mob/living/silicon/robot/robot_model.dm [hunk 3]
```diff
@@ -301,6 +301,27 @@
...
+		/obj/item/plant_analyzer,
+		/obj/item/gun/energy/floragun,
+		/obj/item/cultivator,
+		/obj/item/borg/apparatus/beaker,
+		/obj/item/secateurs,
+		/obj/item/hatchet,
+		/obj/item/shovel/spade,
+		/obj/item/extinguisher/mini)
+	emag_modules = list(
+	)//none for now lmao
```
Comment (Fikou) [inline code/modules/mob/living/silicon/robot/robot_model.dm:320]: then dont put the variable in at all

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 4]
```diff
@@ -805,6 +805,15 @@
 	construction_time = 120
 	category = list("Cyborg Upgrade Modules")
 
+/datum/design/borg_transform_hydroponics
+	name = "Cyborg Upgrade (Botany Model)"
+	id = "borg_transform_hydroponics"
+	build_type = MECHFAB
+	build_path = /obj/item/borg/upgrade/transform/hydroponics
+	materials = list(/datum/material/iron = 15000, /datum/material/glass = 15000, /datum/material/uranium = 1000)
+	construction_time = 120
```
Comment (Fikou) [inline code/modules/research/designs/mechfabricator_designs.dm:814]: use time defines

**Review by Fikou — CHANGES_REQUESTED** [review]

**Comment by timothymtorres** [comment]
> he deserves a flower hat.

This.  Also give the borg the blue hydroponics apron.  This is long overdue and honestly should be one of the roundstart borg modules.

**Comment by Moltovz** [comment]
> > he deserves a flower hat.
> 
> This. Also give the borg the blue hydroponics apron. This is long overdue and honestly should be one of the roundstart borg modules.
image

**Comment by ATH1909** [comment]
I say this as someone who loves standard borgs to death: good god, that sprite is ugly

> These borgs assist botanist. They can NOT do the job all by themself!
they're fuckin' useless, then, especially since they're not even available at roundstart

just give service borgs a plant analyzer and a seed extractor bag, then let drink apparatuses seeds (and maybe plants with an upgrade). boom, done.

**Comment by tralezab** [comment]
yeah like all of this could simply be an upgrade on service borgs but for some reason it isn't

**Comment by Sightld2** [comment]
I feel like the standard practice for borgs is that they can do everything their named module does. The philosophy so far seems to me that the borg should be able to do everything in its niche role better than a human, but the cost being that they lack versatility and can do nothing else. For example an upgraded mediborg can do everything a doctor can and better in most cases, except for maybe CPR.  At the very least, i don't see why it shouldn't be able to harvest and do its job on par with regular botanists

**Comment by Toastgoats** [comment]
I like the idea of having more things borgs can do but this seems like it requires a disproportionate amount of work for how much benefit it provides. 

Imho borgs are beat suited towards playing a really active role in maintaining the station (engieborgs can quickly repair the station while building is limited, janiborgs can clean around the station very quickly) so tying one to a single job seems needlessly shortsighted, especially when it's not necessarily vital to the station and can easily be done by other roles when needed.

I think the idea is really cute but I feel like it should be rolled into service borgs or something else entirely.

**Comment by Sightld2** [comment]
>tying one to a single job seems shortsighted, especially when it's not necessarily vital to the station and can easily be done by other roles when needed.
> 
> I think the idea is really cute but I feel like it should be rolled into service borgs or something else entirely.
> 
> (Tldr botany isn't broad enough to warrant a borg model)

I feel like its justified being so niche, because its unlike the round start modules in that its a researched upgrade that would be the second since clown. I think it makes sense for future nonstandard borg modules to be more niche while the options available are the most productive to the station and its maintenance.

**Comment by Toastgoats** [comment]
I agree with niche borg models being locked behind research, but I think midround borg models should be designed to deal with the problems that come with it.

An exclusively hydroponics borg wouldn't really help much on a midround station because
1: It'd be redundant because botany is already producing food on their own and using the automatic maintenance feature on the trays if needed.
2: It'd be useless because there's no botanists and the borg can't do botany on its own.

I feel like at it's most useful it'd be babysitting a single room.

**Comment by Sightld2** [comment]
> I agree with niche borg models being locked behind research, but I think midround borg models should be designed to deal with the problems that come with it.
> 
> An exclusively hydroponics borg wouldn't really help much on a midround station because 1: It'd be redundant because botany is already producing food on their own and using the automatic maintenance feature on the trays if needed. 
This is a good point, i definitely see botany being pretty established by the time this is researched.

2: It'd be useless because there's no botanists and the borg can't do botany on its own.
> 
> I feel like at it's most useful it'd be babysitting a single room.
This is true as well which is why i'd like it to be able to fully fulfill its niche and operate without organic botanists. All other borg modules have everything their job needs to do exactly within them.

_Maybe I'm just a silicon main who wants to do plants yknow?_

**Comment by Sightld2** [comment]
> An exclusively hydroponics borg wouldn't really help much on a midround station because 1: It'd be redundant because botany is already producing food on their own and using the automatic maintenance feature on the trays if needed. 2: It'd be useless because there's no botanists and the borg can't do botany on its own.

True botany would probably have plenty of time to be well established by the time these are researched. And as well I don't like that the borg can't even do the full job of a botanist, all other borgs have everything they need to preform their job exactly. I feel like if it were changed to be fully capable of botany, it would be useful in situations where botany is understaffed as you mentioned, and allow for _certain silicon mains_ who so desire, to grow plants.

**Comment by Toastgoats** [comment]
Actually this gave me an idea
Have this be an upgrade for _mediborgs_

I'm talking stick a hydroponics tray on a mediborg and let it grow and inject the contents of certain healing plants, plus some botany tools for maintaining it's head plant. Plus a tool for dealing with kudzu.

That way it has a much more proactive role around the station beyond babysitting the hydroponics trays.

This is probably a terrible idea that'd be hell to code but it sounds fun.

**Comment by squi1-rasb** [comment]
I love this, the sprite is very pretty :)

**Comment by LukasBeedellCodestuff** [comment]
Amazing sprite holy shit man.

**Comment by Farquaar** [comment]
I'm indifferent on the idea, but the sprite is... really not great.
Borg sprites shouldn't use the default borg as a base. They should all look unique.

**Comment by SuperNovaa41** [comment]
Sprite looks cool but making an entire cyborg module for an underplayed subdepartment makes no sense

**Comment by bluetshirtguy** [comment]
Looks just like any other borg, laaaame.

**Comment by wraith-54321** [comment]
what does it get when emagged/malf hacked

**Comment by zxaber** [comment]
If the borg can't even do botany completely, I'm not really sure what the point is. At best, this could help robotics in a pinch if a particular botany plant is desperately needed by the station, before being reset back to a more useful model.

**Comment by Sightld2** [comment]
> If the borg can't even do botany completely, I'm not really sure what the point is. At best, this could help robotics in a pinch if a particular botany plant is desperately needed by the station, before being reset back to a more useful model.

As much as I really want a botany borg, I do agree its rather pointless if It can’t completely preform the job. I find it especially strange that the borg has a floral somatory despite being only intended to maintain plants.

**Comment by jlsnow301** [comment]
Just in my op, this did it right:
[#2862](https://github.com/BeeStation/BeeStation-Hornet/pull/2862)

Not a huge fan of the sprite, but I think it could be a skin variation for service borgs

**Comment by IndieanaJones** [comment]
Isnt there already a botany borg sprite in the files made by the original borg artist?

**Comment by Coastfront** [comment]
Few issues with this as others have pointed out.
1. usually cyborgs are meant to be able to do their modules role VERY effectively, but they arent versatile and are meant to be one trick ponies. Engiborgs are extremely good at repairing and building, but cant do literally anything other than engineering. Same with mediborgs, janiborgs, serviceborgs, etc. If the hydroponics borg is just meant to be a botanist except they cant do all of botany and just exist to be slightly helpful, why build one at all?

2. By the time this would be researched botany almost definetly doesent need new help since they've probably gotten everything figured out, as someone else pointed out.

3. Botany in general is just a very very small subdepartment who almost never needs outside help (its rarely staffed by entirly new botanists either, usually there is atleast one experienced botanist there) and if botany had no-one so someone made a borg to supplement that (like when robotics makes a mining borg because all the miners fucked off and died 3 minutes in) the hydroborg cant even help since it doesent do botany fully.

and lastly, no offense, But having it just be standard borg sprite except its covered in flowers and has an apron just doesen't look that good.

**Comment by Dorsisdwarf** [comment]
Ugly, visually confusing and messy sprite, and there's already sprites for a botanyborg that are, if old, at least not just a smoosh of random pixels

**Comment by Mothblocks** [comment]
Large changes like these are expected to have motivating design docs behind them now, with some good explanation on why it's good for the game. Having an eight-word "Why It's Good For The Game" is a very big red flag to me.

**Pull request closed without merging**